### ISSUE TITLE
vpc-cni v1.12.6 and v1.12.5 not working after update to EKS 1.27 if cluster has prometheus-adapter installed

### ISSUE THREAD

**Issue by sirantd** [aws/amazon-vpc-cni-k8s#2393]
**What happened**:
After updating EKS cluster to 1.27 vpc-cni (installed via add-ons) stopped to work. There are no informative logs [debug severity] in /var/log/aws-routed-eni/ipamd.log on the node, and keeps restarting.

Tried several versions (installed manually from the release page) and found that:
1.12.6 - not working
1.12.5 - not working
1.12.2 - working
1.11.5 - working

No IPv6 support enabled for cluster
Additional settings of VPC-CNI:
```
ENABLE_PREFIX_DELEGATION = "true"
WARM_PREFIX_TARGET       = "1"
```

Security groups is not the issue as all the changes performed on the same node, just different vpc-cni versions.

Here is logs when not working:
```
{"level":"info","ts":"2023-05-25T23:55:21.458Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-25T23:55:21.459Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-25T23:55:21.481Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-25T23:55:42.133Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-25T23:55:42.133Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-25T23:55:42.141Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-25T23:55:42.142Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-25T23:55:42.152Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-25T23:56:12.100Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-25T23:56:12.101Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-25T23:56:12.109Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-25T23:56:12.109Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-25T23:56:12.124Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-25T23:56:56.127Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-25T23:56:56.127Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-25T23:56:56.136Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-25T23:56:56.137Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-25T23:56:56.148Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-25T23:58:27.108Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-25T23:58:27.108Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-25T23:58:27.119Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-25T23:58:27.119Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-25T23:58:27.130Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-26T00:01:10.091Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-26T00:01:10.091Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-26T00:01:10.103Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-26T00:01:10.104Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
{"level":"info","ts":"2023-05-26T00:01:10.121Z","caller":"wait/wait.go:222","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"info","ts":"2023-05-26T00:05:17.506Z","caller":"logger/logger.go:52","msg":"Constructed new logger instance"}
{"level":"info","ts":"2023-05-26T00:05:17.509Z","caller":"eniconfig/eniconfig.go:61","msg":"Initialized new logger as an existing instance was not found"}
{"level":"info","ts":"2023-05-26T00:05:17.518Z","caller":"aws-k8s-agent/main.go:28","msg":"Starting L-IPAMD   ..."}
{"level":"info","ts":"2023-05-26T00:05:17.520Z","caller":"aws-k8s-agent/main.go:39","msg":"Testing communication with server"}
```

And after upgrade to the working version:
```
{"level":"info","ts":"2023-05-26T00:05:19.528Z","caller":"wait/wait.go:211","msg":"Successful communication with the Cluster! Cluster Version is: v1.27+. git version: v1.27.1-eks-2f008fe. git tree state: clean. commit: abfec7d7e55d56346a5259c9379dea9f56ba2926. platform: linux/amd64"}
{"level":"warn","ts":"2023-05-26T00:05:19.562Z","caller":"awssession/session.go:64","msg":"HTTP_TIMEOUT env is not set or set to less than 10 seconds, defaulting to httpTimeout to 10sec"}
{"level":"debug","ts":"2023-05-26T00:05:19.567Z","caller":"ipamd/ipamd.go:379","msg":"Discovered region: us-east-1"}
{"level":"info","ts":"2023-05-26T00:05:19.567Z","caller":"ipamd/ipamd.go:379","msg":"Custom networking enabled false"}
{"level":"debug","ts":"2023-05-26T00:05:19.567Z","caller":"awsutils/awsutils.go:415","msg":"Found availability zone: us-east-1b "}
{"level":"debug","ts":"2023-05-26T00:05:19.568Z","caller":"awsutils/awsutils.go:415","msg":"Discovered the instance primary IPv4 address: 172.16.39.233"}
{"level":"debug","ts":"2023-05-26T00:05:19.568Z","caller":"awsutils/awsutils.go:415","msg":"Found instance-id: i-04ece13897592ae92 "}
{"level":"debug","ts":"2023-05-26T00:05:19.569Z","caller":"awsutils/awsutils.go:415","msg":"Found instance-type: t4g.medium "}
{"level":"debug","ts":"2023-05-26T00:05:19.570Z","caller":"awsutils/awsutils.go:415","msg":"Found primary interface's MAC address: 02:5b:88:ee:f2:93"}
{"level":"debug","ts":"2023-05-26T00:05:19.570Z","caller":"awsutils/awsutils.go:415","msg":"eni-00aa95a4332192cde is the primary ENI of this instance"}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"awsutils/awsutils.go:415","msg":"Found subnet-id: subnet-0de276a77839b4579 "}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:388","msg":"Using WARM_ENI_TARGET 1"}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:390","msg":"Using MINIMUM_IP_TARGET 1"}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:391","msg":"Using WARM_PREFIX_TARGET 1"}
{"level":"info","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:409","msg":"Prefix Delegation enabled false"}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:414","msg":"Start node init"}
{"level":"debug","ts":"2023-05-26T00:05:19.571Z","caller":"ipamd/ipamd.go:446","msg":"Max ip per ENI 5 and max prefixes per ENI 0"}
{"level":"info","ts":"2023-05-26T00:05:19.572Z","caller":"awsutils/awsutils.go:1643","msg":"Will attempt to clean up AWS CNI leaked ENIs after waiting 4m41s."}
{"level":"info","ts":"2023-05-26T00:05:19.573Z","caller":"ipamd/ipamd.go:456","msg":"Setting up host network... "}
{"level":"debug","ts":"2023-05-26T00:05:19.573Z","caller":"networkutils/network.go:280","msg":"Trying to find primary interface that has mac : 02:5b:88:ee:f2:93"}
{"level":"debug","ts":"2023-05-26T00:05:19.576Z","caller":"networkutils/network.go:280","msg":"Discovered interface: lo, mac: "}
...
```

**Environment**:
- Kubernetes version (use `kubectl version`): Server Version: version.Info{Major:"1", Minor:"27+", GitVersion:"v1.27.1-eks-2f008fe", GitCommit:"abfec7d7e55d56346a5259c9379dea9f56ba2926", GitTreeState:"clean", BuildDate:"2023-04-14T20:40:28Z", GoVersion:"go1.20.3", Compiler:"gc", Platform:"linux/amd64"}
- CNI Version: 1.12.6
- OS (e.g: `cat /etc/os-release`):
```
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```
- Kernel (e.g. `uname -a`): Linux ip-172-16-39-233.ec2.internal 5.10.178-162.673.amzn2.aarch64 #1 SMP Mon Apr 24 23:34:12 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux

**Comment by jmriebold** [comment]
Currently seeing the same thing on a cluster I just upgraded to 1.27.

**Comment by orsenthil** [comment]
Could you share the steps you tried.

1.  I updated my cluster from EKS 1.26 to EKS 1.27.
2.  Installed the vpc cni addon

```
$ aws eks create-addon --cluster-name senthilx-08May --addon-name vpc-cni --addon-version v1.12.6-eksbuild.2 --resolve-conflicts overwrite
{
    "addon": {
        "addonName": "vpc-cni",
        "clusterName": "senthilx-08May",
        "status": "CREATING",
        "addonVersion": "v1.12.6-eksbuild.2",
        "health": {
            "issues": []
        },
        "addonArn": "arn:aws:eks:us-west-2:428051789465:addon/senthilx-08May/vpc-cni/e0c42ada-8da2-7866-4e53-a149650a183c",
        "createdAt": "2023-05-25T18:38:32.535000-07:00",
        "modifiedAt": "2023-05-25T18:38:32.574000-07:00",
        "tags": {}
    }
}
```
3. Verified the status

```
$ aws eks describe-addon --cluster-name senthilx-08May --addon-name vpc-cni
{
    "addon": {
        "addonName": "vpc-cni",
        "clusterName": "senthilx-08May",
        "status": "ACTIVE",
        "addonVersion": "v1.12.6-eksbuild.2",
        "health": {
            "issues": []
        },
        "addonArn": "arn:aws:eks:us-west-2:428051789465:addon/senthilx-08May/vpc-cni/e0c42ada-8da2-7866-4e53-a149650a183c",
        "createdAt": "2023-05-25T18:38:32.535000-07:00",
        "modifiedAt": "2023-05-25T18:39:05.104000-07:00",
        "tags": {}
    }
}
```

**Comment by jmriebold** [comment]
@orsenthil have you checked the status of the pods themselves? In my case they were crashlooping endlessly like this

```
Installed /host/opt/cni/bin/aws-cni
Installed /host/opt/cni/bin/egress-v4-cni
time="2023-05-25T16:20:47Z" level=info msg="Starting IPAM daemon... "
time="2023-05-25T16:20:47Z" level=info msg="Checking for IPAM connectivity... "
time="2023-05-25T16:20:48Z" level=info msg="Copying config file... "
time="2023-05-25T16:20:48Z" level=info msg="Successfully copied CNI plugin binary and config file."
time="2023-05-25T16:20:48Z" level=error msg="Failed to wait for IPAM daemon to complete" error="exit status 2"
```

but IPAM logs showed no errors.

**Comment by sirantd** [comment]
1. Had a cluster EKS 1.26 with 4 addons installed (including VPC-CNI v.12.5)
2. Updated to EKS 1.27 and found that `aws-node` pods keeps restarting
3. Initiated update (via web console) to the v.12.6.eks.2 - same result
4. Installed `kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.12.2/config/master/aws-k8s-cni.yaml` to make it work

Yes, the same errors in pod logs as @jmriebold provided above

**Comment by orsenthil** [comment]
I couldn't see that.


```
$ kubectl get pods -n kube-system |grep aws-node
aws-node-77nlr             1/1     Running   0            41m
aws-node-l2glx             1/1     Running   0            41m
aws-node-mbvmq             1/1     Running   0            41m
aws-node-rmbtf             1/1     Running   0            41m
aws-node-w8phq             1/1     Running   0            41m
```

And
```
$ kubectl logs aws-node-77nlr -n kube-system -f
Defaulted container "aws-node" out of: aws-node, aws-vpc-cni-init (init)
Installed /host/opt/cni/bin/aws-cni
Installed /host/opt/cni/bin/egress-v4-cni
time="2023-05-26T01:39:00Z" level=info msg="Starting IPAM daemon... "
time="2023-05-26T01:39:00Z" level=info msg="Checking for IPAM connectivity... "
time="2023-05-26T01:39:01Z" level=info msg="Copying config file... "
time="2023-05-26T01:39:01Z" level=info msg="Successfully copied CNI plugin binary and config file."

```
---- 

> time="2023-05-25T16:20:48Z" level=error msg="Failed to wait for IPAM daemon to complete" error="exit status 2"

This error indicates ipamd daemon crashed during update. We can try to see what happened in your scenario. 

This command is expected work successfully when you use addons in your cluster.

> aws eks create-addon --cluster-name senthilx-08May --addon-name vpc-cni --addon-version v1.12.6-eksbuild.2 --resolve-conflicts overwrite

**Comment by sirantd** [comment]
> This command is expected work successfully when you use addons in your cluster.
> 
> > aws eks create-addon --cluster-name senthilx-08May --addon-name vpc-cni --addon-version v1.12.6-eksbuild.2 --resolve-conflicts overwrite

```
❯ aws eks create-addon --cluster-name MY_CLUSTER_NAME --addon-name vpc-cni --addon-version v1.12.6-eksbuild.2 --resolve-conflicts overwrite

An error occurred (ResourceInUseException) when calling the CreateAddon operation: Addon already exists.
```

**Comment by jmriebold** [comment]
We're using a self-managed VPC CNI daemonset (via the [official Helm chart](https://github.com/aws/eks-charts/tree/master/stable/aws-vpc-cni)), but I tried both v1.12.6 and v1.12.6-eksbuild.2 image tags to no avail.

Worth noting that, like @sirantd, we have prefix delegation enabled.

**Comment by orsenthil** [comment]
@sirantd , you will have remove the old addon, before trying new.

> Worth noting that, like @sirantd, we have prefix delegation enabled.

Will try this scenario. cc: @jdn5126

**Comment by sirantd** [comment]
Heh, advanced parameters are not a case... I should think about it earlier but I can't believe it can be related...

Problem solved by removing `prometheus-adapter` which is used to create some custom metrics.
```
❯ helm -n monitoring uninstall prometheus-adapter
release "prometheus-adapter" uninstalled
```

More details you can find here: https://github.com/helm/helm/issues/12061#issuecomment-1537935360 (issue I had also, but didn't know they have the same roots).

https://github.com/kubernetes/kubernetes/pull/116603

So: `The fix is to upgrade client-go to v0.27.x (or v0.26.4 which has the fix backported)`

**Comment by sirantd** [comment]
One more piece of evidence https://github.com/kubernetes-sigs/prometheus-adapter/issues/581

@jmriebold could you confirm that you have prometheus-adapter in your cluster installed?

**Comment by jmriebold** [comment]
@sirantd I do indeed! Excellent detective work. So if I'm understanding correctly, this issue is a combination of Kubernetes 1.27, Prometheus Adapter, and a specific range of client-go versions?

**Comment by sirantd** [comment]
> @sirantd I do indeed! Excellent detective work. So if I'm understanding correctly, this issue is a combination of Kubernetes 1.27, Prometheus Adapter, and a specific range of client-go versions?

So, from my understanding `prometheus-adapter` generates a bad response. At some point, go-client library started to crash parsing it. It has been fixed but requires a lot of projects to rebuild using the patched library version. Or, maintainers of `prometheus-adapter` could be so kind as to issue a new version with a fix. Of course, some other components could also generate bad responses but they are more rarely used so I don't know about them.

**Comment by jmriebold** [comment]
Interesting. I wouldn't have thought VPC CNI could possibly have any contact with Prometheus Adapter.

**Comment by jdn5126** [comment]
Awesome debugging work @sirantd ! The IPAM daemon does run a metrics server at startup: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/aws-k8s-agent/main.go#L71, and if that crashes from a bad response, it does look it would take the IPAM daemon and the `aws-node` pod down with it. That runs before the RPC server is started, so the logging lines up.

To resolve this on our end, it sounds like we need to update our `client-go` version. Note that if you do not need prometheus metrics, the metrics server can be disabled by setting `DISABLE_METRICS` environment variable. I also want to investigate how we can tolerate this failure without taking down the IPAM daemon.

**Comment by sirantd** [comment]
Thanks you @jdn5126 for the suggestion `DISABLE_METRICS`. I tested it, and it's working like a charm with installed `prometheus-adapter`. It gives us time until a new version of `vpc-cni` is released.

**Comment by orsenthil** [comment]
@sirantd - Thank you for debugging and sharing this outcome.

Yes, this seems like an issue with promethus-adapter + kubernetes 1.27 which can block the metrics endpoint.  I could reproduce it at my end

- Not using promethus-adapter, which I understand is for _custom metrics_ until the bug fixed.
- Or as Jeff pointed out `DISABLE_METRICS` might be best to avoid this.

--- 

> it sounds like we need to update our client-go version
Not sure if this will be solve it. It is interesting me that old versions of CNI didn't see this issue.

> I also want to investigate how we can tolerate this failure without taking down the IPAM daemon.

Yes, it will be good idea. This the metrics endpoint (due to custom metrics from promethus-adapter) is blocking and isn't letting the ipamd start to complete.

**Comment by jdn5126** [comment]
The client-go upgrade that will fix this crash merged in https://github.com/aws/amazon-vpc-cni-k8s/pull/2396 and will ship in v1.13.1, which is planned for the end of this month

**Comment by chris-codaio** [comment]
The EKS hosted add-on for `vpc-cni` won't allow a 1.27 kube cluster to downgrade to a version that works.  Any chance to include a working version in the set supported by EKS or publish a working version of the add-on sooner than end of month?

**Comment by chris-codaio** [comment]
> @chris-codaio you should be able to downgrade to `v1.12.5-eksbuild.2` with EKS 1.27, and that version has a `client-go` without the issue: https://github.com/aws/amazon-vpc-cni-k8s/blob/v1.12.5/go.mod#L39
> 
> I can see that version with `aws eks describe-addon-versions --addon-name vpc-cni --kubernetes-version 1.27` and can install it with `aws eks update-addon --cluster-name $CLUSTER_NAME --addon-name vpc-cni --addon-version v1.12.5-eksbuild.2 ...`

Ah, thanks for clarifying.  I had read that all flavors of `v1.12.5` were broken.

**Comment by nicraMarcin** [comment]
I have the same issue and I cannot downgrade it from terraform
supported version fot eks 1.27
```shell
ADDONS	vpc-cni	networking
ADDONVERSIONS	v1.12.6-eksbuild.2
COMPATIBILITIES	1.27	True
ADDONVERSIONS	v1.12.6-eksbuild.1
COMPATIBILITIES	1.27	False
ADDONVERSIONS	v1.12.5-eksbuild.2
COMPATIBILITIES	1.27	False
ADDONVERSIONS	v1.12.5-eksbuild.1
ADDONVERSIONS	v1.12.2-eksbuild.1
```

when I set cluster_addnons to v1.12.2
```
cluster_addons = {
    vpc-cni = {
      # addon_version     = "v1.12.2-eksbuild.1"
      resolve_conflicts = "OVERWRITE"
      most_recent       = false
    }
  }
```
EKS says that is impossible

```
│ Error: updating EKS Add-On (sportsbook-kuka:vpc-cni): InvalidParameterException: Addon version specified is not supported
│ {
│   RespMetadata: {
│     StatusCode: 400,
│     RequestID: "aa4e2ec0-db9c-4664-8582-5728fd4ac9a8"
│   },
│   AddonName: "sportsbook-kuka",
│   ClusterName: "vpc-cni",
│   Message_: "Addon version specified is not supported"
│ }
│ 
│   with aws_eks_addon.this["vpc-cni"],
│   on main.tf line 382, in resource "aws_eks_addon" "this":
│  382: resource "aws_eks_addon" "this" {
│ 
```

**Comment by danijelk** [comment]
> I have the same issue and I cannot downgrade it from terraform supported version fot eks 1.27
> 
> ```shell
> ADDONS	vpc-cni	networking
> ADDONVERSIONS	v1.12.6-eksbuild.2
> COMPATIBILITIES	1.27	True
> ADDONVERSIONS	v1.12.6-eksbuild.1
> COMPATIBILITIES	1.27	False
> ADDONVERSIONS	v1.12.5-eksbuild.2
> COMPATIBILITIES	1.27	False
> ADDONVERSIONS	v1.12.5-eksbuild.1
> ADDONVERSIONS	v1.12.2-eksbuild.1
> ```
> 
> when I set cluster_addnons to v1.12.2
> 
> ```
> cluster_addons = {
>     vpc-cni = {
>       # addon_version     = "v1.12.2-eksbuild.1"
>       resolve_conflicts = "OVERWRITE"
>       most_recent       = false
>     }
>   }
> ```
> 
> EKS says that is impossible
> 
> ```
> │ Error: updating EKS Add-On (sportsbook-kuka:vpc-cni): InvalidParameterException: Addon version specified is not supported
> │ {
> │   RespMetadata: {
> │     StatusCode: 400,
> │     RequestID: "aa4e2ec0-db9c-4664-8582-5728fd4ac9a8"
> │   },
> │   AddonName: "sportsbook-kuka",
> │   ClusterName: "vpc-cni",
> │   Message_: "Addon version specified is not supported"
> │ }
> │ 
> │   with aws_eks_addon.this["vpc-cni"],
> │   on main.tf line 382, in resource "aws_eks_addon" "this":
> │  382: resource "aws_eks_addon" "this" {
> │ 
> ```

use 12.2.5, that one doesn't have the issue?

**Comment by nicraMarcin** [comment]
> use 12.2.5, that one doesn't have the issue?

This is not compatible with eks 1.27

**Comment by danijelk** [comment]
> > use 12.2.5, that one doesn't have the issue?
> 
> This is not compatible with eks 1.27

It is.

![image](https://github.com/aws/amazon-vpc-cni-k8s/assets/580753/91d0ae60-d7a5-4516-a59d-1baf2a4aa419)

![image](https://github.com/aws/amazon-vpc-cni-k8s/assets/580753/2ef28fab-a9b4-4083-b97f-feeb196ea6ab)

**Comment by nicraMarcin** [comment]
so `aws eks describe-addon-versions --addon-name vpc-cni` says that it isn't :thinking: I'll check it.

**Comment by danijelk** [comment]
> so `aws eks describe-addon-versions --addon-name vpc-cni` says that it isn't 🤔 I'll check it.

yeah your output is confusing as it clearly states false on compatibility, but luckily it worked for us to downgrade in Gui, until a proper fix is released end of month.

**Comment by nicraMarcin** [comment]
@danijelk Thank you, it seems that version `v1.12.5-eksbuild.2` works fine and doesn't have this issue.

I tested also with `v1.12.6-eksbuild.2` disabling default rules in prometheus-adapter
```yaml
rules:
  default: false
``` 
and also works.

Thank you

**Comment by jdn5126** [comment]
Closing as this is fixed in v1.13.2 release
